### ISSUE TITLE
Maven Plugin should output `.nondex` directory in the project's build output by default

### ISSUE THREAD

**Issue by bdemers** [TestingResearchIllinois/NonDex#178]
The maven plugin currently litters the project tree with a `.nondex` fold in each module of a multi-module project.

The default output directory _should_ be the project's build directory (`target` by default).

**Comment by kaiyaok2** [comment]
`.nondex` folder is where `NonDex` stores temporary files for the `debug` mojo (if one wants a stack trace of the invocation of an under-determined Java API). We have a `clean` mojo to remove those: for example, with `NonDex 2.1.1`, run:
`mvn edu.illinois:nondex-maven-plugin:2.1.1:clean`

**Comment by bdemers** [comment]
@kaiyaok2 right! And that's a good idea, but littering a project's source directory is not _typical_ behavior for a Maven plugin; any generated content _should_ go into the project build output-dir.
